tests/tests/test_host_kdump.sh

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/bash
+#
+# Copyright 2024 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 3, as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+if [ ! -v DEVICE_IP ]; then
+    DEVICE_IP="192.168.102.125"
+fi
+
+echo "Installing linux-crashdump and rebooting"
+ssh ubuntu@$DEVICE_IP sudo apt install linux-crashdump
+if [ $? != 0 ]; then
+    echo "Can't install linux-crashdump"
+    exit -1
```

Comment:
I wonder how useful it is to use custom shell return codes for each failure type here? Could we instead just use the return code from the ssh command?

Reply:
I'm not sure its ever useful (error codes in bash scripts), but I like to give someone the option of figuring out where it failed given a return code.  I think often people will either throw and exception in these cases with a message only (in python).  I don't think the return code from ssh is useful to figure out where it failed.  I think these things are dilemmas.  I prefer this way, but I certainly understand why an exception would be great.  In this case we'll definitely get any error codes in stdout.

Reply:
I changed my mind and made this return -1.

Reply:
Note that exit codes are unsigned, so this is the same as exit 255. fwiw, my completely unsolicited opinion is that exiting with the `ssh` error code feels like the typical approach. In fact, I suggest using `set -e` if at all possible.

Reply:
Thanks Dan, makes sense